core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -345,8 +345,9 @@ object HistoryServer extends Logging {
     // occur from the keytab.
     if (conf.get(History.KERBEROS_ENABLED)) {
       // if you have enabled kerberos the following 2 params must be set
-      val principalName = conf.get(History.KERBEROS_PRINCIPAL)
-        .getOrElse(throw new NoSuchElementException(History.KERBEROS_PRINCIPAL.key))
+      val principalName = SparkHadoopUtil.get.getServerPrincipal(conf
```

Comment:
How thrit server handle this?

Reply:
This PR is only for handling _HOST at  the History Server . Therefore this change won't impact thrift server